src/zenml/repository.py

```diff
@@ -1096,6 +1107,27 @@ def get_pipeline(
             RuntimeError: If no stack name is specified and no active stack name
                 is configured.
         """
+        if pipeline and pipeline_name:
+            raise TypeError(
+                "'pipeline' and 'pipeline_name' both set for the"
+                "get_pipeline() method. This is not supported. "
+                "Please set either the 'pipeline' or the "
+                "'pipeline_name' parameter."
+            )
+        elif pipeline:
+            if isinstance(
```

Comment:
Same as the comment on the `get_step`, this should probably use `issubclass(...)` if the behavior is intended and also needs an update on the type annotation